Lib/test/test_email/test_message.py

```diff
@@ -1054,6 +1055,37 @@ def test_string_payload_with_multipart_content_type(self):
         attachments = msg.iter_attachments()
         self.assertEqual(list(attachments), [])
 
+    def test_string_payload_with_base64_cte(self):
+        msg = message_from_string(textwrap.dedent("""\
+        Content-Transfer-Encoding: base64
+
+        SGVsbG8uIFRlc3Rpbmc=
+        """), policy=policy.default)
+        self.assertEqual(msg.get_payload(decode=True), b"Hello. Testing")
+        self.assertDefectsEqual(msg['content-transfer-encoding'].defects, [])
+
+    def test_string_payload_with_extra_space_after_cte(self):
+        # https://github.com/python/cpython/issues/98188
+        cte = "base64 "
+        msg = message_from_string(textwrap.dedent(f"""\
+        Content-Transfer-Encoding: {cte}
+
+        SGVsbG8uIFRlc3Rpbmc=
+        """), policy=policy.default)
+        self.assertEqual(msg.get_payload(decode=True), b"Hello. Testing")
+        self.assertDefectsEqual(msg['content-transfer-encoding'].defects, [])
+
+    def test_string_payload_with_extra_text_after_cte(self):
+        msg = message_from_string(textwrap.dedent("""\
+        Content-Transfer-Encoding: base64 some text
+
+        SGVsbG8uIFRlc3Rpbmc=
+        """), policy=policy.default)
+        self.assertEqual(msg.get_payload(decode=True), b"Hello. Testing")
+        cte = msg['content-transfer-encoding']
+        self.assertDefectsEqual(cte.defects, [InvalidHeaderDefect])
```

Comment:
These tests do no belong here as (a) they are not directly testing the EmailMessage or MIMEPart classes and (b) they are not specific to MIMEPart, which is what the tests in this class are.  (Note: there was originally only one unique MIMEPart test, the test test_string_payload_with_multipart_content_type that was added above is *also* in the wrong place, but that's not something to address in this PR).

Instead, I would place these tests at the end of the TestNonConformant test class in test_email.

Reply:
If I remember correctly, the email module accepts as much as possible and uses defects to record errors instead of raising an error.

‘Some text’ is not a valid mechanism. It will be parsed as `some` because no spaces are allowed. However, 'some' might be a valid mechanism according to the RFC: `mechanism := ietf-token / x-token`,

> Currently, the decoded payload would not be decoded at all if I give "base64 blabla", right?

Yes